### PR TITLE
RPM message simulator (dummy publisher)

### PR DESCRIPTION
**Pull request by roman-dvorak**
Hi, this PR contains a simple application for publishing RPM message with custom value. It is useful for the development of the additional application that requires RPM message. 

Usage:
```
rpm_simulator <rpm value>
```

nsh example:
```
pxh> listener rpm
never published

pxh> rpm_simulator 23.45
INFO  [rpm_simulator] RPM message with RPM=23.450 was published

pxh> listener rpm
TOPIC: rpm
 rpm_s
	indicated_frequency_rpm: 23.4500
	estimated_accurancy_rpm: 0.2345

pxh> rpm_simulator
INFO  [rpm_simulator] Usage: rpm_simulator <published RPM>
INFO  [rpm_simulator] Exit. Without publishing any message.

pxh> 
```

### CODE REVIEW

File: src/drivers/rpm/rpm_simulator/rpm_simulator.c [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		PX4_INFO("Usage: rpm_simulator <published RPM>");
+		PX4_INFO("Exit. Without publishing any message.");
+		return 0;
+	}
+
+	struct rpm_s rpm;
+
+	memset(&rpm, 0, sizeof(rpm));
+
+	orb_advert_t rpm_pub = orb_advertise(ORB_ID(rpm), &rpm);
```
Comment (dagar) [inline src/drivers/rpm/rpm_simulator/rpm_simulator.c:73]:
```suggestion
	uORB::Publication<rpm_s> rpm_pub{ORB_ID(rpm)};
```

File: src/drivers/rpm/rpm_simulator/rpm_simulator.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+	// prpepare RPM data message
+	rpm.timestamp = timestamp_us;
+
+	rpm.indicated_frequency_rpm = frequency;
+
+	rpm.estimated_accurancy_rpm = frequency / 100.0f;
+
+	// publish data
+	orb_publish(ORB_ID(rpm), rpm_pub, &rpm);
```
Comment (dagar) [inline src/drivers/rpm/rpm_simulator/rpm_simulator.c:87]:
```suggestion
	rpm_pub.publish(rpm);
```

File: src/drivers/rpm/rpm_simulator/rpm_simulator.c [hunk 3]
```diff
@@ -0,0 +1,92 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2012-2020 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/rpm/rpm_simulator/rpm_simulator.c:3]:
```suggestion
 *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```

File: src/drivers/rpm/rpm_simulator/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,39 @@
+############################################################################
+#
+#   Copyright (c) 2015 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/rpm/rpm_simulator/CMakeLists.txt:3]:
```suggestion
#   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```

File: src/drivers/rpm/rpm_simulator/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+############################################################################
+px4_add_module(
+	MODULE examples__rpm_simulator
+	MAIN rpm_simulator
+	SRCS
+		rpm_simulator.c
```
[the remark this answers is not in the document]
Comment (roman-dvorak) [reply src/drivers/rpm/rpm_simulator/CMakeLists.txt:37]: Done

File: src/drivers/rpm/rpm_simulator/rpm_simulator.c [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+	rpm.timestamp = timestamp_us;
+
+	rpm.indicated_frequency_rpm = frequency;
+
+	rpm.estimated_accurancy_rpm = frequency / 100.0f;
+
+	// publish data
+	orb_publish(ORB_ID(rpm), rpm_pub, &rpm);
+
+	PX4_INFO("RPM message with RPM=%.3f was published", (double)frequency);
```
[the remark this answers is not in the document]
Comment (roman-dvorak) [reply src/drivers/rpm/rpm_simulator/rpm_simulator.c:89]: Wow.. this is great tool, used

File: src/drivers/rpm/rpm_simulator/rpm_simulator.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (roman-dvorak) [reply src/drivers/rpm/rpm_simulator/rpm_simulator.c]: Ok, transfered

**Comment by roman-dvorak** [comment]
Hi,
@dagar thank you for reviewing. This application was created according to [PX4 dev guide](https://dev.px4.io/master/en/apps/hello_sky.html#subscribing-to-sensor-data) where C non-objective access is used.
Thanks

**Comment by roman-dvorak** [comment]
I'm not sure why SITL tests fails. Please, can you give me some advice? Does it require some new tests for rpm_simulator? 
I'm able to run SITL with gazebo.

**Comment by TSC21** [comment]
> I'm not sure why SITL tests fails. Please, can you give me some advice? Does it require some new tests for rpm_simulator?
> I'm able to run SITL with gazebo.

It's an unrelated fail. It's been intermitent since some days. We are looking into it.

**Comment by roman-dvorak** [comment]
Please, @dagar, are there any other requirements? I think they're all fixed.

File: src/drivers/rpm/rpm_simulator/rpm_simulator.cpp [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+	// check input
+	if (argc != 2) {
+		PX4_INFO("Usage: rpm_simulator <published RPM>");
+		PX4_INFO("Exit. Without publishing any message.");
+		return 0;
+	}
+
+	rpm_s rpm{};
+	memset(&rpm, 0, sizeof(rpm));
```
Comment (dagar) [inline src/drivers/rpm/rpm_simulator/rpm_simulator.cpp:63]:
```suggestion
```

**Comment by dagar** [comment]
It looks like a NuttX submodule change slipped in. Can you fix that and I'll merge?

**Comment by roman-dvorak** [comment]
It is now merged with upstream/master

**Comment by roman-dvorak** [comment]
I'm not sure why some tests were canceled.

**Comment by dagar** [comment]
The overall diff is still showing the NuttX submodule change, and now Tools/sitl_gazebo has changed as well.

**Comment by roman-dvorak** [comment]
@dagar now it looks. that it is possible to merge.

File: src/modules/mavlink/mavlink_messages.cpp [hunk 8]
```diff
@@ -90,6 +90,7 @@
 #include <uORB/topics/sensor_combined.h>
 #include <uORB/topics/sensor_gyro_integrated.h>
 #include <uORB/topics/sensor_gyro_status.h>
+#include <uORB/topics/rpm.h>
```
Comment (dagar) [inline src/modules/mavlink/mavlink_messages.cpp:93]:
```suggestion
```

**Pull request merged**